I managed to catch one exception that I've seen in production as well:
```
2019-12-05 22:19:23,302:INFO:WorkQueueBackend:Going to delete 1 documents in *workqueue* db for workflow: amaltaro_SC_MultiPU_Dec2019_Val_191205_200132_476. Doc IDs: [u'ca2b6170b120a241148fd4d042b4e3f2']
2019-12-05 22:19:23,382:INFO:WorkQueueBackend:Going to delete 1 documents in *workqueue_inbox* db for workflow: amaltaro_TaskChain_RelVal_Multicore_Dec2019_Val_191205_200206_4257. Doc IDs: [u'amaltaro_TaskChain_RelVal_Multicore_Dec2019_Val_191205_200206_4257']
2019-12-05 22:19:23,435:INFO:WorkQueueBackend:Going to delete 2 documents in *workqueue_inbox* db for workflow: amaltaro_SC_MultiPU_Dec2019_Val_191205_200132_476. Doc IDs: [u'ca2b6170b120a241148fd4d042b4e3f2', u'amaltaro_SC_MultiPU_Dec2019_Val_191205_200132_476']
2019-12-05 22:19:23,459:ERROR:WorkQueue:Error deleting WQ elements: AMR j doc: {u'key': u'ca2b6170b120a241148fd4d042b4e3f2', u'error': u'not_found'}
Traceback (most recent call last):
  File "/data/srv/HG1912c/sw/slc7_amd64_gcc630/cms/workqueue/1.2.9.pre3/lib/python2.7/site-packages/WMCore/WorkQueue/WorkQueue.py", line 999, in performQueueCleanupActions
    res = self.deleteCompletedWFElements()
  File "/data/srv/HG1912c/sw/slc7_amd64_gcc630/cms/workqueue/1.2.9.pre3/lib/python2.7/site-packages/WMCore/WorkQueue/WorkQueue.py", line 922, in deleteCompletedWFElements
    return self.backend.deleteWQElementsByWorkflow(deleteRequests)
  File "/data/srv/HG1912c/sw/slc7_amd64_gcc630/cms/workqueue/1.2.9.pre3/lib/python2.7/site-packages/WMCore/WorkQueue/WorkQueueBackend.py", line 588, in deleteWQElementsByWorkflow
    couchdb.bulkDeleteByIDs(docIds)
  File "/data/srv/HG1912c/sw/slc7_amd64_gcc630/cms/workqueue/1.2.9.pre3/lib/python2.7/site-packages/WMCore/Database/CMSCouch.py", line 682, in bulkDeleteByIDs
    raise Exception("AMR j doc: %s" % j)
Exception: AMR j doc: {u'key': u'ca2b6170b120a241148fd4d042b4e3f2', u'error': u'not_found'}
```

we try to delete the same document first in workqueue, then in workqueue_inbox. The one in workqueue_inbox cannot be found and that's why we get the `KeyError` exception.
I don't really get why it would complain though, given that it's a different database...

Nonetheless, I'm going to add a check there to avoid having these exceptions.

--- a/src/python/WMCore/Database/CMSCouch.py
+++ b/src/python/WMCore/Database/CMSCouch.py
@@ -674,6 +674,9 @@ class Database(CouchDBRequests):
         docs = self.allDocs(keys=ids)['rows']
         for j in docs:
             doc = {}
+            if "id" not in j:
+                print("Document not found: %s" % j)
+                continue
             doc["_id"] = j['id']
             doc["_rev"] = j['value']['rev']
             self.queueDelete(doc)

--- a/src/python/WMCore/GlobalWorkQueue/CherryPyThreads/CleanUpTask.py
+++ b/src/python/WMCore/GlobalWorkQueue/CherryPyThreads/CleanUpTask.py
@@ -1,5 +1,6 @@
 from __future__ import (division, print_function)
 
+from time import time
 from WMCore.REST.CherryPyPeriodicTask import CherryPyPeriodicTask
 from WMCore.WorkQueue.WorkQueue import globalQueue
 
@@ -22,8 +23,9 @@ class CleanUpTask(CherryPyPeriodicTask):
         2. synchronize cancelled elements.
         We can also make this in the separate thread
         """
-
+        start = int(time())
         globalQ = globalQueue(**config.queueParams)
         globalQ.performQueueCleanupActions(skipWMBS=True)
-
+        end = int(time())
+        self.logger.info("%s executed in %d secs.", self.__class__.__name__, end - start)
         return

--- a/src/python/WMCore/WorkQueue/WorkQueue.py
+++ b/src/python/WMCore/WorkQueue/WorkQueue.py
@@ -911,12 +911,14 @@ class WorkQueue(WorkQueueBase):
 
         # fetch workflows known to workqueue + workqueue_inbox and with spec attachments
         reqNames = self.backend.getWorkflows(includeInbox=True, includeSpecs=True)
+        self.logger.info("Retrieved %d workflows known by WorkQueue", len(reqNames))
         requestsInfo = self.requestDB.getRequestByNames(reqNames)
         deleteRequests = []
         for key, value in requestsInfo.items():
             if (value["RequestStatus"] is None) or (value["RequestStatus"] in deletableStates):
                 deleteRequests.append(key)
-
+        self.logger.info("Found %d out of %d workflows in a deletable state",
+                         len(deleteRequests), len(reqNames))
         return self.backend.deleteWQElementsByWorkflow(deleteRequests)
 
     def performSyncAndCancelAction(self, skipWMBS):
@@ -987,6 +989,7 @@ class WorkQueue(WorkQueueBase):
         msg = 'Finished elements: %s\nCanceled workflows: %s' % (', '.join(["%s (%s)" % (x.id, x['RequestName']) \
                                                                             for x in finished_elements]),
                                                                  ', '.join(wf_to_cancel))
+        self.logger.info(msg)
         self.backend.recordTaskActivity('housekeeping', msg)
 
     def performQueueCleanupActions(self, skipWMBS=False):

--- a/src/python/WMCore/WorkQueue/WorkQueueBackend.py
+++ b/src/python/WMCore/WorkQueue/WorkQueueBackend.py
@@ -575,14 +575,20 @@ class WorkQueueBackend(object):
         options["stale"] = "update_after"
         options["reduce"] = False
 
+        idsByWflow = {}
         for couchdb in dbs:
             result = couchdb.loadView("WorkQueue", "elementsByWorkflow", options, workflowNames)
-            ids = []
             for entry in result["rows"]:
-                ids.append(entry["id"])
-            if ids:
-                couchdb.bulkDeleteByIDs(ids)
-                deleted += len(ids)
+                idsByWflow.setdefault(entry['key'], [])
+                idsByWflow[entry['key']].append(entry['id'])
+            for wflow, docIds in idsByWflow.items():
+                self.logger.info("Going to delete %d documents in *%s* db for workflow: %s. Doc IDs: %s",
+                                 len(docIds), couchdb.name, wflow, docIds)
+                try:
+                    couchdb.bulkDeleteByIDs(docIds)
+                except CouchNotFoundError as exc:
+                    self.logger.error("Failed to find one of the documents. Error: %s", str(exc))
+                deleted += len(docIds)
         # delete the workflow with spec from workqueue db
         for wf in workflowNames:
             self.db.delete_doc(wf)

